drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -322,100 +530,99 @@ impl Transceivers {
             if !mask.is_empty() {
                 let fpga = self.fpga(fpga_index);
                 let wdata: U16<byteorder::LittleEndian> = U16::new(mask.get());
-                fpga.write(op, addr, wdata)?;
+                // mark that an error occurred so we can modify the success mask
+                if fpga.write(op, addr, wdata).is_err() {
+                    match fpga_index {
+                        FpgaController::Left => error |= LEFT_LOGICAL_MASK,
```

Comment:
Does it make sense to set the error for the entire logical mask associated with the FPGA?

e.g. if I provide `LogicalPortMask(1)` and the `fpga.write` fails, having 16 error bits in the returned mask could be confusing.

Reply:
Never mind; you mask it out below!